### ISSUE TITLE
[clang-tidy] Add IgnoreMacros option to bugprone-lambda-function-name

### ISSUE THREAD

**Issue by carlosgalvezp** [llvm/llvm-project#62857]
Hi!

We have code like this triggering the `bugprone-lambda-function-name`:

```cpp
#define MY_ASSERT(x) foo(__func__)

void foo(char const* func);

int main()
{
    [](int x) { MY_ASSERT(x); }(123);
}
```
Godbolt: https://godbolt.org/z/GqddPvWed

It would be good if an option `IgnoreMacros` could be added to disable the warning in these scenarios. The warning is a true positive and would display the wrong message, but given also file and number information in the assert it should not cause much trouble - at least not more trouble than code filled with `NOLINT`s.
